Vision Pro Surgery Project (Initial Version)

AdritRao opened the pull request:
# Vision Pro Surgery Project (Initial Version)

## :gear: Release Notes 
Consists of the first version of the **server-side** and **Vision Pro** applications. Server app is developed using Python for both Mac and Windows OS (with packaged executables). Enables user to port their computer with surgical video towers and efficiently start a local network video stream. The server app reveals the port ID and IP address which the user can connect to through the Vision Pro application.

```README.md``` file is updated with an initial version of documentation that includes a set up guide and more project details. 

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

AdritRao committed:
Version 1

First version of streaming and Vision Pro app.

AdritRao committed:
Update README.md

AdritRao committed:
Add files via upload

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
Update README.md

AdritRao committed:
Add files via upload

AdritRao committed:
Rename SpeziServer2.png to SpeziServerv2.png

AdritRao committed:
Delete README_assets/SpeziServerv2.png

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
Update CONTRIBUTORS.md

AdritRao committed:
Update README.md

AdritRao committed:
Delete README_assets/VP_v1.png

AdritRao committed:
Delete README_assets/VP_v2.png

AdritRao committed:
Add files via upload

AdritRao committed:
Update README.md

AdritRao committed:
SwiftLint fixes

AdritRao committed:
Merge branch 'v1' of https://github.com/StanfordBDHG/VisionProSurgery into v1

AdritRao committed:
App Icon

PSchmiedmayer commented:
@vishnuravi I might not get to this before the end of the winter break. 

@AdritRao is working on fixing the SwiftLint errors and other build errors in the next few weeks & look at some of the build errors to get them into a passing state.
@AdritRao Feel free to remove some of the GitHub Action jobs that are not really applicable to this app, e.g., watchOS ones that are left over from the Template App 👍

PSchmiedmayer commented on .github/workflows/build-and-test.yml:
@AdritRao Feel free to remove some of the GitHub Action jobs that are not really applicable to this app, e.g., watchOS ones that are left over from the Template App 👍

PSchmiedmayer replied:
Would be amazing that we can adjust all the CI GitHub Actions to this project to ensure that we have a proper CI setup for this project.

PSchmiedmayer reviewed (commented):
@vishnuravi I might not get to this before the end of the winter break. 

@AdritRao is working on fixing the SwiftLint errors and other build errors in the next few weeks & look at some of the build errors to get them into a passing state.

PSchmiedmayer commented on .periphery.yml:
```diff
@@ -1,16 +1,16 @@
-- TemplateApplication
-- TemplateApplicationTests
-- TemplateApplicationUITests
-- TemplateWatchApplication
+- VisionProSurgery
+- VisionProSurgeryTests
+- VisionProSurgeryUITests
+- VisionProSurgeryWatchApplication
```
I think the targets configuration is outdated.

PSchmiedmayer commented on CONTRIBUTORS.md:
```diff
@@ -1,14 +1,16 @@
+Vision Pro Surgery Contributors
 =================================
 
+* [Adrit Rao](https://github.com/AdritRao)
```
🚀

PSchmiedmayer commented on README.md:
Very cool job with the README!

PSchmiedmayer commented on SpeziServerApp/Executable/Spezi Server:
Wondering if we should commit the compiled binary here in the app; might make more sense to attach it as an Artifact to an eventually release but maybe only contain the code within the repo? We should add very nice scripts to easily generate them locally.

PSchmiedmayer commented on SpeziServerApp/spezi_server.py:
Would be great if we can move some of the Spezi Data Pipeline Python linting infrastructure here into this repo as some GitHub Actions.

PSchmiedmayer commented on VisionProSurgery/Assets.xcassets/AppIcon.appiconset/appstore.png:
Nice job with the icon!

PSchmiedmayer commented on VisionProSurgery/Resources/Localizable.xcstrings:
```diff
@@ -0,0 +1,147 @@
+
+    },
+    "ACCESSIBILITY_ICON_LABEL" : {
+      "extractionState" : "manual",
```
We might want to enable automatic extraction for all strings in this localization catalogue.

PSchmiedmayer commented on VisionProSurgery/SetUp.swift:
Nice job with the setup; might be good to separate the views into individual files and add some previews here to make it easier to maintain and review in Xcode. Also applies to some of the other views and files here.

PSchmiedmayer commented on create.sh:
I think that file can be removed; IMO just there to easily rename the template application but not needed once you are done with the rename.

PSchmiedmayer commented on VisionProSurgeryUITests/VisionProSurgeryUITests.swift:
We should add some minimal UI tests for the application to ensure that future contributions are easier to review & setup.

PSchmiedmayer commented on VisionProSurgery/WebView.swift:
```diff
@@ -0,0 +1,51 @@
+
+import SwiftUI
+import WebKit
+
```
Generally want to add two empty lines between the import and types. Also applies to elements between content elements in types as suggested in this file.
```suggestion


```

PSchmiedmayer commented on VisionProSurgery/WebView.swift:
```diff
@@ -0,0 +1,51 @@
+    @State private var width: Int = 640
+    @State private var height: Int = 480
+    @State private var fps: Int = 20
+
```
```suggestion


```

PSchmiedmayer commented on VisionProSurgery/WebView.swift:
```diff
@@ -0,0 +1,51 @@
+        .frame(width: 640, height: 480)
+    }
+}
+
```
```suggestion


```

PSchmiedmayer commented on VisionProSurgery/WebView.swift:
```diff
@@ -0,0 +1,51 @@
+                Spacer()
+            }
+        }
+        .frame(width: 640, height: 480)
```
```suggestion
            .frame(width: 640, height: 480)
```

PSchmiedmayer commented on VisionProSurgery/WebView.swift:
```diff
@@ -0,0 +1,51 @@
+
+struct WebView: UIViewRepresentable {
+    let url: URL
+
```
```suggestion


```

PSchmiedmayer commented on VisionProSurgery/WebView.swift:
```diff
@@ -0,0 +1,51 @@
+        uiView.load(request)
+    }
+}
+
```
```suggestion


```

PSchmiedmayer reviewed (changes requested):
Nice job with the initial setup @AdritRao!!

Very cool! I had a few comments thought the project & some suggestions around formatting & setting the project up with a good linting & CI setup to allow us to easily maintain this in the future. Let me know if you have any questions 🚀

Exciting to get closer to a small user validation study; let's sync up on this in the next few days.

AdritRao committed:
Update README.md

AdritRao committed:
Update VisionProSurgery/WebView.swift

Co-authored-by: Paul Schmiedmayer <PSchmiedmayer@users.noreply.github.com>

AdritRao committed:
Addressing PR comments

AdritRao committed:
Delete watchOS

AdritRao committed:
Fixing periphery

AdritRao committed:
Add Vision Pro icon

AdritRao committed:
Delete iOS icons

AdritRao committed:
Add exe script

AdritRao committed:
Update README.md

AdritRao committed:
Update build-and-test.yml

AdritRao committed:
Merge branch 'v1' of https://github.com/StanfordBDHG/VisionProSurgery into v1

AdritRao committed:
Added CI

AdritRao committed:
Added UI testing (updated version 2)

AdritRao committed:
Add icon to view

AdritRao committed:
Fix periphery

AdritRao committed:
Python project structure

AdritRao committed:
Fix paths for linting

AdritRao committed:
Fix lint issues and add comments

AdritRao committed:
Lint formatting

AdritRao committed:
linting

AdritRao committed:
Lint

AdritRao committed:
Swift lint

adritrao commented:
Hi @PSchmiedmayer, thank you for reviewing the PR!

Main updates made:
1. Added a script (```create_exe.py```) to create a cross-platform executable from the ```spezi_server.py``` (instead of uploading the executable directly to GitHub).
2. Added Python lint infrastructure (based on Spezi Data Pipeline). 
3. Fixed all Swift lint and Pylint formatting issues.

Looking forward to the upcoming initial validation study.

PSchmiedmayer committed:
Fix REUSE, Lint, and CodeQL

PSchmiedmayer committed:
Fix REUSE, Lint, and CodeQL

PSchmiedmayer commented on README.md:
Great README; looks amazing!

PSchmiedmayer committed:
Update Project Setup

PSchmiedmayer committed:
Remove Build Plugin

PSchmiedmayer committed:
Update Assets

PSchmiedmayer commented on VisionProSurgery/SetUp.swift:
```diff
@@ -0,0 +1,201 @@
+
+import SwiftUI
+
+struct PortInputView: View {
```
Fine to merge it as is; but in the future we might want to extract https://github.com/StanfordSpezi/SpeziAccount/pull/88/files#diff-588430140cebb8c27170e8a89e9cbecf47cb43d9f00df90b2aa3eb6d43d912d8 into SpeziViews and use it here; has a lot of sophisticated logic to jump between the text views and other elements that are making data entry easier. Might be good to add an GitHub issue for this.

PSchmiedmayer reviewed (approved):
Thank you for all the work here; happy to see this merged.

I fixed the remaining GitHub Actions in this PR 🚀
